### PR TITLE
Overlap and normalization corrections

### LINKED ISSUE #18: Issue with floatrange intersection
Hello,

There seems to be a bug with floatrange intersection (or maybe I do not understand it correctly).
It can be observed when first range has same upper value as the lower value of the second range, but lower value of the second range is not included.

Here is an example:
`floatrange(0.0, 5.0, upper_inc=True).intersection(floatrange(5.0, lower_inc=False))`

Which results in:
`floatrange(5.0, 5.0, lower_inc=False, upper_inc=True)`

As I understand, expected result should be an empty set, because 5.0 is not included in the first range, so it can also not be an intersection.
The result is also misleading, because same number is both included and not included (a Schrödinger range?).

### PR DESCRIPTION
**Pull request by lgharibashvili**
Fixes #18

### CODE REVIEW

File: tests/test_range.py [hunk 1]
```diff
@@ -36,7 +37,9 @@ def test_bound_helper():
     assert _Bound(1, inc=False, is_lower=False) < _Bound(1, inc=True, is_lower=True)
     assert _Bound(1, inc=False, is_lower=False) < _Bound(1, inc=True, is_lower=False)
 
-    assert _Bound(1, inc=False, is_lower=True) < _Bound(1, inc=True, is_lower=False)
+    # In this case first bound is a little greater than 1,
+    # the second is exactly 1, so A !< B
+    # assert _Bound(1, inc=False, is_lower=True) < _Bound(1, inc=True, is_lower=False)
```
Comment (runfalk) [inline tests/test_range.py:42]: The comment is right. I think we should swap to the sides and keep the assert. No need to have this comment I think (though I appreciate that you put it ther for the review).

**Review by runfalk — CHANGES_REQUESTED** [review]
Very nicely done. I'll try to merge this in asap and make a new point release. Please poke me again if I haven't done so within a week or two.

File: tests/test_range.py (hunk 1 above)
Comment (lgharibashvili) [reply tests/test_range.py:42]: Will you be able to correct it in the merge or should I change it?

Comment (runfalk) [reply tests/test_range.py:42]: If you could just amend your commit that would be great.

Comment (lgharibashvili) [reply tests/test_range.py]: Updated in the new commit

**Pull request merged**